File: 2022/day1/day1.ts
import { readFileSync } from 'fs';
import {join} from 'path';

const part1 = (input) => {
    return readFileSync(join(__dirname, input), 'utf-8')
        .split('\r\n\r\n')
        .map((val) => 
            val.split('\n')
            .map((elfCals) => 
                Number(elfCals)).reduce((agg, cal) => agg+cal, 0));
}

const ans = part1('in1.txt').sort((a,b)=>b-a)
console.log(ans[0])
console.log(ans.splice(0, 3).reduce((agg, val) => agg+val,))

File: 2022/day1/day1.py
def get_input(file):
    file = open(file, 'r')
    inputNums = []
    totCals = 0
    for line in file:
        line = line.strip()
        if line == "":
            inputNums.append(totCals)
            totCals = 0
            continue
        totCals += int(line)
    return inputNums

def part1(input):
    return max(input)

def part2(input):
    input.sort(reverse=True)
    return input[0] + input[1] + input[2]

def main():
    in1 = "in1.txt"
    in2 = "in2.txt"
    input = get_input(in1)
    print(part1(input))
    print(part2(input))

if __name__ == '__main__':
    main()
